### PR TITLE
CI: Publish tags to Dockerhub

### PR DESCRIPTION
**Pull request by bndw**
Old PR: https://github.com/envoyproxy/envoy/pull/1743

Signed-off-by: bndw <benjamindwoodward@gmail.com>

### CODE REVIEW

**Comment by bndw** [comment]
We're blocked by the DCO check, which I've opened an issue for investigation over at https://github.com/probot/dco/issues/39

**Comment by bndw** [comment]
@mattklein123 DCO check is resolved and all checks have passed (except the mac build which timed out?)

Lmk if there's anything else on this one.

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
        - run: ci/do_circle_ci.sh bazel.release
        - setup_remote_docker
        - run: ci/docker_push.sh
+       - deploy:
```
Comment (mattklein123) [inline .circleci/config.yml]: any reason to use deploy here vs. run?

Comment (bndw) [reply .circleci/config.yml]: deploy was what the examples used, but run will work and read better.

**Comment by bndw** [comment]
Fixes https://github.com/envoyproxy/envoy/issues/1742

**Comment by bndw** [comment]
Sounds good, just need to make sure the tag follows the existing convention (prefixed with `v`) for the job to be run.

**Pull request merged**